protocols/mdns/src/behaviour.rs

```diff
@@ -164,10 +194,19 @@ impl Mdns {
         self.discovered_nodes.iter().map(|(p, _, _)| p)
     }
 
+    fn reset_timer(&mut self) {
+        self.timeout.set_interval(self.query_interval);
+    }
+
+    fn fire_timer(&mut self) {
+        self.timeout
+            .set_interval_at(Instant::now(), self.query_interval);
+    }
+
     fn inject_mdns_packet(&mut self, packet: MdnsPacket, params: &impl PollParameters) {
         match packet {
             MdnsPacket::Query(query) => {
-                self.timeout.set_interval(self.query_interval);
+                self.reset_timer();
```

Comment:
this removes the jitter again, doesn’t it?

Reply:
no, the jitter is computed during mdns initialization. I don't think it makes much difference if it's different every single time

Reply:
computed but not stored in `self.query_interval`, as commented above